test/e2e/must_gather_operator_tests.go

```diff
@@ -1,114 +1,1260 @@
+
+			ginkgo.GinkgoWriter.Printf("✅ Non-admin user successfully deleted MustGather CR: %s\n", mustGatherName)
+		})
+
+		ginkgo.It("should clean up Job and Pod when MustGather CR is deleted", func() {
+			mustGatherName := fmt.Sprintf("test-cascading-delete-%d", time.Now().Unix())
+
+			ginkgo.By("Creating MustGather CR")
+			mg := createMustGatherCR(mustGatherName, ServiceAccount)
+
+			ginkgo.By("Waiting for Job to be created")
+			Eventually(func() error {
+				return adminClient.Get(testCtx, client.ObjectKey{
+					Name:      mustGatherName,
+					Namespace: testNamespace,
+				}, &batchv1.Job{})
+			}).WithTimeout(2 * time.Minute).WithPolling(5 * time.Second).Should(Succeed())
+
+			ginkgo.By("Deleting MustGather CR")
+			err := nonAdminClient.Delete(testCtx, mg)
+			Expect(err).NotTo(HaveOccurred())
+
+			ginkgo.By("Verifying Job is eventually cleaned up")
+			Eventually(func() bool {
+				err := adminClient.Get(testCtx, client.ObjectKey{
+					Name:      mustGatherName,
+					Namespace: testNamespace,
+				}, &batchv1.Job{})
+				return apierrors.IsNotFound(err)
+			}).WithTimeout(2*time.Minute).WithPolling(5*time.Second).Should(BeTrue(),
+				"Job should be cleaned up when MustGather CR is deleted")
+		})
 	})
 })
+
+// Helper Functions
+
+func createNonAdminClient() client.Client {
+	ginkgo.By("Creating impersonated non-admin client")
+	nonAdminConfig := rest.CopyConfig(adminRestConfig)
+	nonAdminConfig.Impersonate = rest.ImpersonationConfig{
+		UserName: nonAdminUser,
+	}
+
+	c, err := client.New(nonAdminConfig, client.Options{Scheme: testScheme})
+	Expect(err).NotTo(HaveOccurred(), "Failed to create non-admin impersonation client")
+
+	ginkgo.GinkgoWriter.Printf("✅ Created impersonated client for user: %s\n", nonAdminUser)
+	return c
+}
+
+func verifyOperatorDeployment() {
+	ginkgo.By("Checking must-gather-operator namespace exists")
+	ns := &corev1.Namespace{}
+	err := adminClient.Get(testCtx, client.ObjectKey{Name: "must-gather-operator"}, ns)
+	Expect(err).NotTo(HaveOccurred(), "must-gather-operator namespace should exist")
+
+	ginkgo.By("Checking must-gather-operator deployment exists and is available")
+	deployment := &appsv1.Deployment{}
+	err = adminClient.Get(testCtx, client.ObjectKey{
+		Name:      "must-gather-operator",
+		Namespace: "must-gather-operator",
+	}, deployment)
+	Expect(err).NotTo(HaveOccurred(), "must-gather-operator deployment should exist")
+
+	// Verify deployment has at least one ready replica
+	Expect(deployment.Status.ReadyReplicas).Should(BeNumerically(">", 0),
+		"must-gather-operator deployment should have at least one ready replica")
+
+	ginkgo.GinkgoWriter.Printf("✅ must-gather-operator is deployed and available (ready replicas: %d)\n",
+		deployment.Status.ReadyReplicas)
+}
+
+func createTestNamespace() {
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: testNamespace,
+			Labels: map[string]string{
+				"test": "non-admin-must-gather",
```

Comment:
we should add another generic label eg. operator.openshift.io/must-gather-operator: e2e-test, etc.